rfc-178-update-asset-lifecycle-via-publishing-api.md

```diff
@@ -0,0 +1,184 @@
+For new assets, we COULD provide the necessary auth bypass ID and access limiting organisation IDs with the original asset, but it would be preferable not to do that if we can avoid it. Ideally the only path to setting this data would be via Publishing API. To enable this, we COULD create a new piece of state in Asset Manager which prevents serving the asset until a first update to the edition is processed.
+
+After this, publishing applications must continue to include sufficient information about attachments and images in Publishing API payloads to allow Asset Manager to make appropriate state updates. Attachments sent to Publishing API by Whitehall look like this:
+
+```json
+{
+    "accessible": false,
+    "alternative_format_contact_email": "different.format@hmrc.gov.uk",
+    "attachment_type": "file",
+    "command_paper_number": "",
+    "content_type": "application/vnd.oasis.opendocument.text",
+    "file_size": 11349,
+    "filename": "Amendments_5_to_8_to_Clause_37-Claim_for_relief_on_foreign_income.odt",
+    "hoc_paper_number": "",
+    "id": "8485786",
+    "isbn": "",
+    "title": "Amendments 5 to 8 to Clause 37: Claim for relief on foreign income",
+    "unique_reference": "",
+    "unnumbered_command_paper": false,
+    "unnumbered_hoc_paper": false,
+    "url": "https://assets.publishing.service.gov.uk/media/67bf19a4750837d7604dbb96/Amendments_5_to_8_to_Clause_37-Claim_for_relief_on_foreign_income.odt"
+}
+```
+
+Asset Manager's ID for the asset is part of the attachment URL, but we MAY provide it as a separate field so that Asset Manager can easily perform a lookup for the correct asset. We MUST add a deletion marker and the replacement ID (if replaced) to the attachment data to allow Asset Manager to make the correct state adjustments for attachment assets. Deleted and replaced attachments MUST be filtered out of the data sent to content store.
+
+Images look like this:
+
+```json
+{
+  "alt_text": "The launch of the Amelia Troubridge photography exhibition at Getty Images Gallery, London.",
+  "caption": "The launch of the Amelia Troubridge photography exhibition at Getty Images Gallery, London.",
+  "high_resolution_url": "https://assets.publishing.service.gov.uk/government/uploads/system/uploads/image_data/file/65710/s960_Fanzi-Down.jpg",
+  "url": "https://assets.publishing.service.gov.uk/government/uploads/system/uploads/image_data/file/65710/s300_Fanzi-Down.jpg"
+}
+```
+
+Again, we MAY include the Asset Manager IDs for each image, and we MUST provide deletion and replacement state. Deleted and replaced images MUST be filtered out of the data sent to content store.
+
+Publishing API already places a message on a RabbitMQ exchange each time an edition is published. We MUST add another topic for updates to draft editions so that Asset Manager can make additional updates to access limiting and access bypass states for draft documents. This may also be useful in the future to provide draft stack capability for Search API.
```

Comment:
May be worth linking to https://trello.com/c/5nN8vKG7/600-create-draft-search-index-to-enable-draft-search-preview here